test/libyul/yulOptimizerTests/unusedStoreEliminator/mstore_overwriting_mcopy.yul

```diff
@@ -0,0 +1,28 @@
+{
+    calldatacopy(0, 0, 0x40)
+
+    let _0 := 0
+    let _32 := 32
+    let _42 := 42
+
+    mcopy(_0, _32, _32) // Redundant. MSTORE overwrites it.
```

Comment:
So I guess most of these tests where `mcopy` is redundant and should thus be removed (but are currently not), are only there to fail once we introduce the proper mechanism in `UnusedStoreEliminator` (and likely the `DataFlowAnalyzer`) to deal with this? (And I guess as a 'sanity check' that it doesn't do anything crazy.)

Reply:
Yes, but note that there are very few of those. Like, I can see only 3 among all the tests in `unusedStoreEliminator/`. I did not cover this case extensively exactly because it should not happen. Most of the tests check the impact of `MCOPY` on removal of other instructions.